### ISSUE TITLE
Add export feature to logger

### ISSUE THREAD

**Issue by bogachenko** [uBlockOrigin/uBlock-issues#334]
I want to see the preservation of the log of the network request log in TXT format

**Comment by bogachenko** [comment]
@gwarser Wow, is that all ??

**Comment by gwarser** [comment]
>No description provided.

Is that all?

**Comment by bogachenko** [comment]
@gwarser what in the title of the topic is not clear?

**Comment by bogachenko** [comment]
in English it is written ... I want to see the preservation of the log of the network request log in TXT format

**Comment by bogachenko** [comment]
briefly and clearly. what else?

**Comment by gorhill** [comment]
> I want to see the preservation of the log of the network request log in TXT format

It helps to be more detailed about what you suggest exactly.

Turns out it is something I also often thought would be useful.

**Comment by gwarser** [comment]
At least tell me if you want it only for one time saving (already recorded) or for dumping log to file in real time.

**Comment by gorhill** [comment]
I will just provide an export feature, saving to file requires user interaction.

**Comment by uBlock-user** [comment]
Having a title doesn't automatically qualify for you to not have to provide some description. Make some effort otherwise it will end up as invalid.

**Comment by bogachenko** [comment]
@gwarser 
> At least tell me if you want it only for one time saving (already recorded) or for dumping log to file in real time.

I do not know. Probably FULL logging. ie (activated function in the settings) which keeps all changes, errors, etc. in TXT

@gorhill 
> I will just provide an export feature, saving to file requires user interaction.

that's not bad either. but FULL sounds better (as for me)

@uBlock-user I'll know now. would explain @gwarser  for newbies

**Comment by uBlock-user** [comment]
No need to explain, the template is there to fill out, specifically the **Description** part in the template but you removed the template, next time don't remove the template but rather fill it in.

**Comment by bogachenko** [comment]
@uBlock-user no problem

**Comment by gorhill** [comment]
This will be an export feature in the logger, I also had entertained the idea in the past because too often people end up (surely tediously) screenshotting the logger output when they are asked to provide it. Live download is not possible, anything that write to disk from the extension API must be done as a download operation, which requires user interaction.

**Comment by bogachenko** [comment]
@gorhill 
> Live download is not possible

it is sad

**Comment by kulfoon** [comment]
How the feature will differ than:
1_ selecting all text in the logger by pressing CTRL + A 
2_ copying the selected text by pressing CTRL + C
3_ pasting the selected text into something like Notepad
4_ saving as a txt file

Also it's possible to past the log directly into GitHub comment which looks better ( I've also embedded it into a spoiler):

<details not-open>
<summary>LOG</summary>

15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/electron.png
-- | -- | -- | -- | -- | --
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/atom.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/trollface.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/suspect.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/shipit.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/rage4.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/rage3.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/rage2.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/rage1.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/octocat.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/neckbeard.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/hurtrealbad.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/godmode.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/goberserk.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/finnadie.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/feelsgood.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/bowtie.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/basecampy.png
15:41:39 |   |   |   | image | https://assets-cdn.github.com/images/icons/emoji/basecamp.png
15:40:43 |   |   |   | xhr | https://github.com/preview?markdown_unsupported=false&repository=129621088&subject=334&subject_type=Issue
15:40:04 |   |   |   | xhr | https://github.com/preview?markdown_unsupported=false&repository=129621088&subject=334&subject_type=Issue
15:39:18 |   |   |   | xhr | https://github.com/preview?markdown_unsupported=false&repository=129621088&subject=334&subject_type=Issue
15:35:55 |   |   |   | xhr | https://github.com/hovercards?user_id=43844449&subject=issue%3A388865341&current_path=%2FuBlockOrigin%2FuBlock-issues%2Fissues%2F334
15:35:18 |   | \|\|play.google.com/log? | -- | beacon | https://play.google.com/log?hasfast=true&authuser=0&format=json
15:35:18 |   | \|\|play.google.com/log? | -- | beacon | https://play.google.com/log?hasfast=true&authuser=0&format=json
15:35:18 |   | \|\|play.google.com/log? | -- | xhr | https://play.google.com/log?format=json&hasfast=true&authuser=0
15:35:18 |   | \|\|google.*/gen204? | -- | image | https://translate.google.pl/gen204?client=webapp&langidshow=orig=toporne,sl=pl,corrlang=en
15:35:17 |   | \|\|play.google.com/log? | -- | xhr | https://play.google.com/log?format=json&hasfast=true&authuser=0
15:35:16 |   | \|\|play.google.com/log? | -- | xhr | https://play.google.com/log?format=json&hasfast=true&authuser=0
15:35:15 |   | \|\|google.*/gen204? | -- | image | https://translate.google.pl/gen204?client=webapp&ttsaudio=mp3=1,ogg=1,wav=1,vol=1,type=html5
15:35:15 |   | \|\|apis.google.com/js/rpc:shindig_random.js | -- | script | https://apis.google.com/js/rpc:shindig_random.js?onload=init
15:35:15 |   | \|\|apis.google.com/js/rpc:shindig_random.js | -- | script | https://apis.google.com/js/rpc:shindig_random.js?onload=init
15:35:15 |   |   |   | script | https://ssl.gstatic.com/accounts/o/748736246-postmessagerelay.js
15:35:14 |   |   |   | frame | https://accounts.google.com/o/oauth2/postmessageRelay?parent=https%3A%2F%2Ftranslate.google.pl&jsh=m%3B%2F_%2Fscs%2Fabc-static%2F_%2Fjs%2Fk%3Dgapi.gapi.en.CUp85wbT4DI.O%2Frt%3Dj%2Fd%3D1%2Frs%3DAHpOoo-XBQda2DFvo9hxbj_dGnCV84SJMA%2Fm%3D__features__#rpctoken=538224815&forcesecure=1
15:35:14 |   |   |   | image | https://translate.google.pl/favicon.ico
15:35:14 |   | /gen_204? | -- | image | https://www.google.com/gen_204?atyp=i&zx=1544538914679&ogsr=1&ei=G8sPXP3bMoyRmgWS87WABA&ct=6&cad=i&id=203136001&loc=undefined&prid=51&ogd=pl&ogprm=up&ic=-1
15:35:14 |   |   |   | script | https://apis.google.com/_/scs/abc-static/_/js/k=gapi.gapi.en.CUp85wbT4DI.O/m=client/exm=gapi_iframes,googleapis_client,plusone/rt=j/sv=1/d=1/ed=1/rs=AHpOoo-XBQda2DFvo9hxbj_dGnCV84SJMA/cb=gapi.loaded_1
15:35:14 |   |   |   | script | https://apis.google.com/_/scs/abc-static/_/js/k=gapi.gapi.en.CUp85wbT4DI.O/m=gapi_iframes,googleapis_client,plusone/rt=j/sv=1/d=1/ed=1/rs=AHpOoo-XBQda2DFvo9hxbj_dGnCV84SJMA/cb=gapi.loaded_0
15:35:14 |   |   |   | image | https://ssl.gstatic.com/images/icons/material/system_gm/1x/content_copy_black_24dp.png

</details>

**Comment by bogachenko** [comment]
And is it possible to add the same function in umatrix? or is it necessary to raise this question there ?

**Comment by Zungao** [comment]
Could you please consider raising the logger entry limit from 5000 to something bigger?

**Comment by gwarser** [comment]
Note: tab-less indicator is not currently preserved when copying from the logger. This should be handled somehow when you decide to implement this feature.

**Comment by bogachenko** [comment]
By the way, how do you look at removing (by default or when turning on, but not by a button!) Logs? Ahh? Memory saving.

**Comment by gorhill** [comment]
> This should be handled

This was on my mind.

**Comment by gorhill** [comment]
> how do you look at removing

This was also on my mind as I refactor the logger.

**Comment by bogachenko** [comment]
@gorhill so that? do you implement?

**Comment by bogachenko** [comment]
@gorhill And by the way about sore. Why do closed tabs remain in memory?
![default](https://user-images.githubusercontent.com/8535285/50960972-86d5a180-14be-11e9-8eb5-336a472178c7.png)

**Comment by gorhill** [comment]
> And please answer this question as well.

This can happens uBO is not notified by the extension framework when some tabs were closed -- I don't know the reason of this. This is something I have observed a long, long time ago, hence why there is a "janitor" in uBO which will kick in at regular interval (period of some minutes) to find out whether some tab entries are orphaned. These will disappear eventually.
